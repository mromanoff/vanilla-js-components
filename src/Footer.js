import Component from './Component';
import inject from './decorators/inject';

@inject('mmyStore')
class Footer extends Component {
  instanceProperty = 'Footer';

  constructor(props, store) {
    super(props);
    this.state = {...props}; // Spread in the props from our initial state as the state.
    this.store = store;
  }

  template(props) { // the data is passed in as props here.
    let template = `
        <footer>
            <p>${this.instanceProperty} content going here </p>
            <p>store value: ${this.store.a}</p>
            <p>store value: ${this.store.b}</p> 
        </footer>
      `;
    return template;
  }
};

export default Footer;