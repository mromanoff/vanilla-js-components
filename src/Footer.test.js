/**
 * @jest-environment jsdom
 */

import Footer from './Footer';

it('Footer renders correctly', () => {

  document.componentRegistry = {};
  document.nextId = 0;

  const element = document.createElement('div');
  element.setAttribute('data-component', 'footer');
  let component = new Footer({element});
  component.render();

  expect(component).toMatchSnapshot();
});