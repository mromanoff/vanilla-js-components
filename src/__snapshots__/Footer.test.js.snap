// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
Footer {
  "_id": 1,
  "instanceProperty": "Footer",
  "state": Object {
    "element": <div
      data-component="footer"
    >
      
        
      <footer>
        
            
        <p>
          Footer content going here 
        </p>
        
            
        <p>
          store value: 1
        </p>
        
            
        <p>
          store value: 2
        </p>
         
        
      </footer>
      
      
    </div>,
  },
  "store": Object {
    "a": 1,
    "b": 2,
    "title": "MMY Store title",
  },
}
`;
